### ISSUE TITLE
Bug in tracking MSE loss past first epoch

### ISSUE THREAD

**Issue by ereinha** [yu-rp/KANbeFair#8]
train.py code only records first epoch losses on regression tasks like Special numerical functions because MSE loss decreases with training and code is written for accuracy which increases with training.

```
    best_test_metric = 0
    corresponding_train_metric = 0
    fvctimer = Timer()
    for epoch in range(1, args.epochs + 1):
        if fvctimer.is_paused():
            fvctimer.resume()
        else:
            fvctimer.reset()
        train(args, model, device, train_loader, optimizer, epoch, logger = logger, start_index = (epoch - 1) *len(train_loader))
        fvctimer.pause()
        train_metric = test(args, model, device, train_loader, logger = logger, name = "train")
        test_metric = test(args, model, device, test_loader, logger = logger, name = "test")
        if test_metric > best_test_metric:
            best_test_metric = test_metric
            corresponding_train_metric = train_metric
```

**Comment by yu-rp** [comment]
Thank @ereinha for the issue. The code is updated as follows: 
```
if args.loss == "cross_entropy":
    best_test_metric = 0 
elif args.loss == "mse":
    best_test_metric = 1e10 
else:
    raise NotImplementedError
corresponding_train_metric = 0

# Code in between #

if args.loss == "cross_entropy":
    if test_metric > best_test_metric:
        best_test_metric = test_metric
        corresponding_train_metric = train_metric
elif args.loss == "mse":
    if test_metric < best_test_metric:
        best_test_metric = test_metric
        corresponding_train_metric = train_metric
else:
    raise NotImplementedError
```
